test/getiBFT_spec.rb

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env rspec
+require_relative '../src/modules/IscsiClientLib'
+
+describe Yast::IscsiClientLibClass do
+
+  before :each do
+    @iscsilib = Yast::IscsiClientLibClass.new
+    @iscsilib.main() 
+  end
+
+  describe "#getiBFT" do
+    context "when filtering output of 'iscsiadm -m fw'" do
+      it "returns data in form of a map " do
+        @iscsilib.stub(:getFirmwareInfo).
+          and_return("# BEGIN RECORD 2.0-872\n"\
+                     "# END RECORD\n"\
+                     "iface.bootproto = STATIC\n"\
+                     "iface.hwaddress = 00:00:c9:b1:bc:7f\n"\
+                     "iface.initiatorname = iqn.2011-05.com.emulex:eraptorrfshoneport1\n"\
+                     "node.conn[0].address = 172.0.21.6\n"\
+                     "node.conn[0].port = 3260\n"\
+                     "node.name = iqn.1986-03.com.ibm:sn.135061874\n"
+                     )
+
+        ibft_data = @iscsilib.getiBFT()
+
+        expect(ibft_data).to eq(
+                                {"# BEGIN RECORD 2.0-872" => "", 
+                                  "# END RECORD" => "", 
+                                  "iface.bootproto" => "STATIC", 
+                                  "iface.hwaddress" => "00:00:c9:b1:bc:7f", 
+                                  "iface.initiatorname" => "iqn.2011-05.com.emulex:eraptorrfshoneport1", 
+                                  "node.conn[0].address" => "172.0.21.6", 
+                                  "node.conn[0].port" => "3260", 
+                                  "node.name" => "iqn.1986-03.com.ibm:sn.135061874"})
+      end
+    end
+
+    context "when could not get list of targets from firmware" do
+      it "returns an empty map " do
+        @iscsilib.stub(:getFirmwareInfo).
+          and_return("")
+
+        ibft_data = @iscsilib.getiBFT()
+         
+        expect(ibft_data).to eq({})
+      end
+    end
+
+    context "when not on x86 hardware" do
+      it "returns an empty map " do
+        Yast::Arch.stub(:x86_64).and_return(false)
+        Yast::Arch.stub(:i386).and_return(false)      
```

Comment:
two notes:

1) please see that above it is not stubbed, that test is on x86_64, so build for ppc or s390 fails as other contexts return empty map
2) I prefer to stub `Yast::Arch.architecture` method. It is one liner and more explicit which architecture is simulated.